x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformIndexer.java

```diff
@@ -1177,7 +1177,7 @@ protected boolean shouldAuditOnFinish(long completedCheckpoint) {
     }
 
     private RunState determineRunStateAtStart() {
-        if (context.from() != null) {
+        if (context.from() != null && changeCollector.queryForChanges()) {
```

Comment:
On line 1190 below we check if `changeCollector == null`. If it can be `null` on line 1190 then surely it can be `null` here, and that will cause an NPE.

Reply:
Done.